cmd/oras/root/cp.go

```diff
@@ -257,9 +257,15 @@ func prepareCopyOption(ctx context.Context, src oras.ReadOnlyGraphTarget, dst or
 		return opts, nil
 	}
 
-	referrers, err := graph.FindPredecessors(ctx, src, index.Manifests, opts)
-	if err != nil {
-		return opts, err
+	// do breadth first search to find all levels of child referrers
+	var referrers []ocispec.Descriptor
+	for descs := index.Manifests; len(descs) > 0; {
+		foundReferrers, err := graph.FindPredecessors(ctx, src, descs, opts)
```

Comment:
What if the result of `FindPrecessors` contains more than referrers? For example, the OCI store also returns the index referencing the specified manifest as predecessors of the specified manifest.

Reply:
Thanks for catching it.